Modules/getpath.py

```diff
@@ -0,0 +1,739 @@
+# READ VARIABLES FROM config
+# ******************************************************************************
+
+program_name = config.get('program_name')
+home = config.get('home')
+executable = config.get('executable')
+base_executable = config.get('base_executable')
+prefix = config.get('prefix')
+exec_prefix = config.get('exec_prefix')
+base_prefix = config.get('base_prefix')
+base_exec_prefix = config.get('base_exec_prefix')
+ENV_PYTHONPATH = config['pythonpath_env']
+use_environment = config.get('use_environment', 1)
+
+pythonpath = config.get('module_search_paths')
+
+stdlib_dir = None
+platstdlib_dir = None
+
+# ******************************************************************************
+# CALCULATE program_name
+# ******************************************************************************
+
+program_name_was_set = bool(program_name)
+
+if not program_name:
+    try:
+        program_name = config.get('orig_argv', [])[0]
+    except IndexError:
+        pass
+
+if not program_name:
+    program_name = DEFAULT_PROGRAM_NAME
+
+if EXE_SUFFIX and not hassuffix(program_name, EXE_SUFFIX):
+    program_name = program_name + EXE_SUFFIX
+
+
+# ******************************************************************************
+# CALCULATE executable
+# ******************************************************************************
+
+if py_setpath:
+    # When Py_SetPath has been called, executable defaults to
+    # the real executable path.
+    if not executable:
+        executable = real_executable
+
+if os_name == 'nt':
+    # QUIRK: Windows defaults to the real executable path
+    if not executable:
+        executable = real_executable
+
+if os_name == 'darwin':
+    # QUIRK: macOS uses real_executable when program_name would
+    # otherwise require a PATH search.
+    if real_executable and not executable and SEP not in program_name:
+        executable = real_executable
+
+if not executable and SEP not in program_name:
+    # Resolve names against PATH. If program_name has a separator, it
+    # will be made absolute against the current directory below.
+    # NOTE: The use_environment value is ignored for this lookup.
+    # To properly isolate, launch Python with a full path.
+    for p in ENV_PATH.split(DELIM):
+        p = joinpath(p, program_name)
+        if isxfile(p):
+            executable = p
+            break
+
+if not executable and program_name:
+    # Resolve partial path program_name against current directory
+    executable = abspath(program_name)
+    #if not program_name_was_set and not isfile(executable):
+    #    # Issue #7774 (also test_sys.test_executable) requires us to
+    #    # only set sys.executable when it exists, but to still correctly
+    #    # evaluate everything else. So we treat the path we were given
+    #    # as the 'real' executable, but clear out the result.
+    #    real_executable = executable
+    #    executable = None
```

Comment:
This is apparently not the place to do this, because it breaks test_embed everywhere.

Somehow directly setting `argv[0]` in the config to a non-existent path is supposed to work, but having it come in through the real `argv[0]` is supposed to leave `sys.executable` blank (but still figure out where everything is... presumably from CWD?)

I'll have to come back to this later...

Reply:
Turns out, it does use cwd, because it blindly joins everything against `''` if that's what we've inferred it to be.